### PR TITLE
Add custom endpoints for check out and in

### PR DESCRIPTION
**Pull request by cassiewallace**
- [x] Add custom endpoints for users to check books out and in 
- [x] Implement checks on checking in and out
- [x] Improve error messages and responses

### CODE REVIEW

File: books/views.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
+from rest_framework import status, viewsets
+from rest_framework.decorators import action, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
 from books.models import Book
```
Comment (quentinR) [inline books/views.py]: usually try to keep things alphabetized

File: books/views.py [hunk 2]
```diff
@@ -15,4 +18,26 @@ class BookViewSet(viewsets.ModelViewSet):
...
+    def check_out(self, request, pk=None):
+        """Check out a book."""
+        book = self.get_object()
+
+        book.checked_out_by = request.user
+        book.save()
+
+        return Response({'status': 'checked out'})
+
+    @action(detail=True, methods=['post'],
```
Comment (quentinR) [inline books/views.py]: Should the method be DELETE here?

File: books/views.py (hunk 1 above)
Comment (cassiewallace) [reply books/views.py]: that's how i had it before, but [Python](https://www.python.org/dev/peps/pep-0008/#imports) says:

```
Imports should be grouped in the following order:

1. Standard library imports.
2. Related third party imports.
3. Local application/library specific imports.

You should put a blank line between each group of imports.
```

File: books/views.py (hunk 2 above)
Comment (cassiewallace) [reply books/views.py]: I was thinking that I'm not deleting the resource, I'm actually updating the value. WDYT?

**Pull request merged**